components/ILIAS/Mail/classes/class.ilMail.php

```diff
@@ -1262,15 +1272,16 @@ private function sendMimeMail(
         $mailer->Send();
     }
 
-    /**
-     * @param list<string> $attachments
-     */
-    public function saveAttachments(array $attachments): void
+    public function saveAttachments(?ResourceCollectionIdentification $attachments): void
     {
+        if (!is_null($attachments)) {
+            $attachments = $attachments->serialize();
+        }
+
         $this->db->update(
             $this->table_mail_saved,
             [
-                'attachments' => ['clob', serialize($attachments)],
+                'attachments' => ['clob', $attachments],
```

Comment:
All database queries and manipulations for table: `mail_saved` / field: `attachments` must use type `text ` IMO, according to the migration in `components/ILIAS/Mail/classes/Setup/class.ilMailDatabaseUpdateSteps.php`.

```suggestion
                'attachments' => ['text', $attachments],
```

We discussed this above and you responded with a :+1: , but it seems to be still the wrong type :smile: .